### PR TITLE
Add the logic to allow waiting list to vote

### PR DESCRIPTION
**Pull request by mzaatar**
PS: needs some love

### CODE REVIEW

File: components/voting.tsx [hunk 1]
```diff
@@ -97,6 +99,7 @@ export default class Voting extends React.PureComponent<VotingProps, VotingState
       votes: this.readFromStorage(storageKey(this.props, StorageKeys.VOTES)),
     })
     this.votingTopRef = React.createRef()
+    console.log(this.props.waitingListCanVoteWithEmail);
```
Comment (antonjb) [inline components/voting.tsx:102]: Probably don't need this anymore

File: components/voting.tsx [hunk 2]
```diff
@@ -446,24 +447,25 @@ export default class Voting extends React.PureComponent<VotingProps, VotingState
...
                   {' '}
                   <span
                     className="fa fa-question-circle"
                     style={{ cursor: 'pointer', fontSize: '20px' }}
                     title={
                       this.props.ticketNumberHandling === TicketNumberWhileVoting.Optional
-                        ? 'Your vote will have a higher weighting if you optionally supply your ticket # from your ticket confirmation email when getting an attendee ticket.'
-                        : 'To submit a vote you must supply your ticket # from your ticket confirmation email when getting an attendee ticket.'
+                        ? VotingWithOptionalTicketMessage
+                        : `To submit a vote you must supply your ${this.props.waitingListCanVoteWithEmail ? 'either' : ''} ticket # from your ticket confirmation email when getting an attendee ticket${this.props.waitingListCanVoteWithEmail ? 'or your email address that you registered with the waiting list.' : ''}.`
```
Comment (antonjb) [inline components/voting.tsx:462]: this doesn't read quite right.

`To submit a vote you must supply your either ticket #...`

The either should come before your

`To submit a vote you must supply either your ticket #...`

The second check is missing a space which means the text will be right up next to the previous text, like this `getting an attendee ticketor your email address`

File: components/voting.tsx [hunk 3]
```diff
@@ -446,24 +447,25 @@ export default class Voting extends React.PureComponent<VotingProps, VotingState
...
                     }
                     onClick={() =>
                       alert(
                         this.props.ticketNumberHandling === TicketNumberWhileVoting.Optional
-                          ? 'Your vote will have a higher weighting if you optionally supply your ticket # from your ticket confirmation email when getting an attendee ticket.'
-                          : 'To submit a vote you must supply your ticket # from your ticket confirmation email when getting an attendee ticket.',
+                          ? VotingWithOptionalTicketMessage
+                          : `To submit a vote you must supply your ${this.props.waitingListCanVoteWithEmail ? 'either' : ''} ticket # from your ticket confirmation email when getting an attendee ticket${this.props.waitingListCanVoteWithEmail ? 'or your email address that you registered with the waiting list.' : ''}.`
```
Comment (antonjb) [inline components/voting.tsx:468]: Similar here as the comment with spacing and grammar

**Review by antonjb — APPROVED** [review]
Couple of small comments but when they're fixed it looks good to go

**Comment by mzaatar** [comment]
Don't merge

**Pull request closed without merging**